geomesa-accumulo/geomesa-accumulo-tools/src/test/scala/org/locationtech/geomesa/accumulo/tools/export/AccumuloExportCommandTest.scala

```diff
@@ -193,27 +193,34 @@ class AccumuloExportCommandTest extends TestWithFeatureType {
     DelimitedTextConverter.magicParsing(sft.getTypeName, new FileInputStream(file)).toList
 
   def readJson(file: String, sft: SimpleFeatureType): Seq[SimpleFeature] = {
-    WithClose(new FeatureJSON().streamFeatureCollection(new FileInputStream(file))) { iter =>
-      val result = Seq.newBuilder[SimpleFeature]
-      while (iter.hasNext) {
-        val f = iter.next
-        result += ScalaSimpleFeature.create(sft, f.getID, f.getAttributes.toArray: _*)
+    val converter = SimpleFeatureConverter.infer(() => new FileInputStream(file), None, Some(file)) match {
+      case None => ko(s"could not create converter from $file"); null: SimpleFeatureConverter
+      case Some((s, c)) => SimpleFeatureConverter(s, c)
+    }
+    val result = Seq.newBuilder[SimpleFeature]
+    val names = sft.getAttributeDescriptors.asScala.map(_.getLocalName)
+    WithClose(converter.process(new FileInputStream(file))) { features =>
+      features.foreach { f =>
+        val copy = new ScalaSimpleFeature(sft, f.getID)
+        names.foreach(a => copy.setAttribute(a, f.getAttribute(a)))
+        result += copy
       }
-      result.result()
     }
+    result.result()
   }
 
   def readLeaflet(file: String, sft: SimpleFeatureType): Seq[SimpleFeature] = {
     val html = IOUtils.toString(new FileInputStream(file), StandardCharsets.UTF_8)
     val i = html.indexOf("var points = ") + 13
     val json = html.substring(i, html.indexOf(";", i))
-    WithClose(new FeatureJSON().streamFeatureCollection(json)) { iter =>
-      val result = Seq.newBuilder[SimpleFeature]
-      while (iter.hasNext) {
-        val f = iter.next
-        result += ScalaSimpleFeature.create(sft, f.getID, f.getAttributes.toArray: _*)
+    val tmp = Files.createTempFile("gm-export-leaflet", ".json").toFile
+    try {
+      WithClose(new FileWriter(tmp))(IOUtils.write(json, _))
+      readJson(tmp.getAbsolutePath, sft)
```

Comment:
Ok, coffee isn't working.  Can you explain why we need a tmp file here?  I might just need a reminder of the context.

Reply:
the leaflet file has all the html and stuff in it - as a hack, i read in the file, pull out just the json, then write that to a tmp file so it can be read by the other readJson method.